airflow/jobs/scheduler_job_runner.py

```diff
@@ -1066,6 +1068,14 @@ def _run_scheduler_loop(self) -> None:
                         num_finished_events += self._process_executor_events(
                             executor=executor, session=session
                         )
+
+                for executor in self.job.executors:
+                    try:
+                        with create_session() as session:
+                            self._process_task_event_logs(executor._task_event_logs, session)
```

Comment:
Why do we create a new session here instead of using the scheduler session? I'm surprised it's not causing issue in the scheduler

Reply:
good question.  look above, on line 1055.  see it is created there?  so i assume it is no different here, and should not be a problem.

the reason to do it, in my view, is that there i want to not let any errors in "inserting logs" take down the scheduler.  the create_session decorator has try / commit / execpt /rollback logic so it should restore things to a good state if there's any problem. 

with that explanation, what do you think?

Reply:
Makes sense. Scheduler session exited before this new session. It's ok

Reply:
What do you think about moving it into the _process_executor_events method of the scheduler. I think with that, we might not have to iterate over the executors twice(see line 1068

Reply:
i thought it better to do in separate transactions. one matters for task execution and scheduler behavior.  one is just logging events.  maybe i'm am being overly cautious but i was thinking to defend against bad behavior in executor event logging.  i don't have a strong feeling about it.

Reply:
Will resolve for now. Can always move it later